pymc3/step_methods/smc_utils.py

```diff
@@ -78,3 +85,200 @@ def _posterior_to_trace(posterior, variables, model, var_info):
             size += new_size
         strace.record({k: v for k, v in zip(varnames, value)})
     return MultiTrace([strace])
+
+
+def metrop_kernel(
+    q_old,
+    old_tempered_logp,
+    proposal,
+    scaling,
+    accepted,
+    any_discrete,
+    all_discrete,
+    discrete,
+    n_steps,
+    prior_logp,
+    likelihood_logp,
+    beta,
+    ABC,
+):
+    """
+    Metropolis kernel
+    """
+    deltas = np.squeeze(proposal(n_steps) * scaling)
+    for n_step in range(n_steps):
+        delta = deltas[n_step]
+
+        if any_discrete:
+            if all_discrete:
+                delta = np.round(delta, 0).astype("int64")
+                q_old = q_old.astype("int64")
+                q_new = (q_old + delta).astype("int64")
+            else:
+                delta[discrete] = np.round(delta[discrete], 0)
+                q_new = floatX(q_old + delta)
+        else:
+            q_new = floatX(q_old + delta)
+
+        if ABC:
```

Comment:
There is no other way to do this?

Reply:
this comes from the SMC code, not sure why I need to add this...

Reply:
to be clear I add it when I refactor the SMC code, I may try to revisit this.

Reply:
fixed